src/components/TimeSelectInput/index.js

```diff
@@ -0,0 +1,505 @@
+                this.hasPropValue = false;
+                return;
+            }
+            this.isMinutesInputFocused = true;
+            this.minutesInputRef.current.focus();
+        }
+    }
+
+    handleFocusHour() {
+        const { hour } = this.state;
+        this.prevHour = hour || this.prevHour;
+        this.setState({
+            inputFocusedIndex: 0,
+            hour: '',
+        });
+    }
+
+    handleBlurHour() {
+        const { hour } = this.state;
+        const { hour24 } = this.props;
+        if (this.isUpOrDownButtonPressed) {
+            this.isUpOrDownButtonPressed = false;
+            return;
+        }
+        if (this.isMinutesInputFocused) {
+            this.isMinutesInputFocused = false;
+            return;
+        }
+        if (hour === '00' && this.value >= '0' && !hour24) {
+            this.setState({
+                hour: '12',
+            });
+        }
+    }
+
+    handleChangeMinutes(event) {
+        const { minutes } = this.state;
+        const { value } = event.target;
+        const { hour24 } = this.props;
+        let normalizedValue;
+
+        if (isNumber(value)) {
+            if (Number(value) > 60 || this.isUpOrDownKeyPressed) {
+                const newTypedValue = getSingleNewTypedValue(minutes, value);
+                normalizedValue = normalizeMinutes(newTypedValue);
+                this.handleChangeTime({
+                    minutes: normalizedValue,
+                });
+            } else {
+                normalizedValue = normalizeMinutes(value);
+                this.handleChangeTime({
+                    minutes: normalizedValue,
+                });
+            }
+
+            const shouldNotFocusNextInput =
+                Number(normalizedValue) < 6 && (!minutes || this.isUpOrDownKeyPressed);
+
+            if (shouldNotFocusNextInput) {
+                this.isUpOrDownKeyPressed = false;
+                return;
+            }
+            if (!hour24) this.amPmInputRef.current.focus();
+        }
+    }
+
+    handleFocusMinutes() {
+        const { minutes } = this.state;
+        this.prevMinutes = minutes || this.prevMinutes;
+        this.setState({
+            inputFocusedIndex: 1,
+            minutes: '',
+        });
+    }
+
+    handleAmPmChange(value) {
+        this.handleChangeTime({
+            inputFocusedIndex: -1,
+            ampm: value,
+        });
+    }
+
+    hanldeFocusAmPm() {
+        this.setState({ inputFocusedIndex: 2 });
+    }
+
+    handleKeyDown(event) {
+        const { keyCode } = event;
+
+        if (keyCode === RIGHT_KEY) {
+            this.handleRightKeyPressed();
+        }
+        if (keyCode === LEFT_KEY) {
+            this.handleLeftKeyPressed();
+        }
+        if (keyCode === UP_KEY) {
+            this.incrementHandler();
+        }
+        if (keyCode === DOWN_KEY) {
+            this.decrementHandler();
+        }
+        if (keyCode === DELETE_KEY) {
+            this.resetState();
```

Comment:
this is not working properly since the hour input value does not reset, we should add test to this behavior